cpp/include/raft/matrix/batched_rearrange.cuh

```diff
@@ -0,0 +1,87 @@
+/*
```

Comment:
I do also prefer `inplace` over `batched`. While I think these should stay in different files in the `detail` namespace, I'd like to combine them in the public APIs with the non-inplace versions (so for eg. add the inplace scatter to `raft/matrix/scatter.cuh`). I would also like if we kept the signatures as close as possible to the non-inplace variants. 

Also, please expose only the mdspan-based variants. There's no sense exposing new pointer-based functions when that API has already been deprecated.

Reply:
Following up here on [this reply](https://github.com/rapidsai/raft/pull/1445/files/a51966a1d1fbf02e31862556d6f1b1d7e62b276c..fefaac27dcd0994397f717695739464235f7505f#r1224674791) by @tarang-jain. 

@cjnolet, can I summarize your points as follows for the inplace gather, scatter functionality:
- name detail headers `detail/scatter_inplace.cuh, detail/gather_inplace.cuh`; 
- name public headers `scatter.cuh, gather.cuh`
- method signatures should preferably be kept as close to the non-inplace versions
- do not provide pointer based overloads
- **the inplace versions of scatter and gather should be an overload(?)**

Personally, I would prefer the inplace versions to be named `gather_inplace` and `scatter_inplace` instead of an overload of `gather` and `scatter` respectively. This really helps when reading the code. Without it, I would for sure have to look it up in code review etc. In PyTorch, there is the convention of having a trailing underscore in the name, e.g., `scatter_`. This might be an option as well, although this would be the first option.

Reply:
Please let me know the final verdict on the public overloads scatter/gather vs new names for scatter and gather.